contracts/Migrator.sol

```diff
@@ -0,0 +1,226 @@
+
+    // From OGVStaking.sol
+    function unstakeFrom(address staker, uint256[] memory lockupIds) external returns (uint256, uint256);
+
+    // From ExponentialStaking.sol
+    function stake(uint256 amountIn, uint256 duration, address to, bool stakeRewards, int256 lockupId) external;
+}
+
+contract Migrator is Governable {
+    ERC20Burnable public immutable ogv;
+    ERC20Burnable public immutable ogn;
+
+    IStaking public immutable ogvStaking;
+    IStaking public immutable ognStaking;
+
+    // Fixed conversion rate
+    uint256 public constant CONVERSION_RATE = 0.09137 ether;
+
+    uint256 public endTime;
+
+    event TokenExchanged(uint256 ogvAmountIn, uint256 ognAmountOut);
+    event Decommissioned();
+    event LockupsMigrated(address indexed user, uint256[] ogvLockupIds, uint256 newDuration);
+
+    error MigrationAlreadyStarted();
+    error MigrationIsInactive();
+    error MigrationNotComplete();
+    error BalanceMismatch(uint256 ogvAmountIn, uint256 expectedOgnAmountOut, uint256 actualOgnAmountOut);
+    error LockupIdsRequired();
+
+    constructor(address _ogv, address _ogn, address _ogvStaking, address _ognStaking) {
+        ogv = ERC20Burnable(_ogv);
+        ogn = ERC20Burnable(_ogn);
+        ogvStaking = IStaking(_ogvStaking);
+        ognStaking = IStaking(_ognStaking);
+    }
+
+    /**
+     * @notice Starts the migration and sets it to end after
+     *          365 days. Also, approves xOGN to transfer OGN
+     *          held in this contract. Can be invoked only once
+     */
+    function start() external onlyGovernor {
+        if (endTime != 0) {
+            revert MigrationAlreadyStarted();
+        }
+
+        // Max approve
+        ogn.approve(address(ognStaking), type(uint256).max);
+
+        endTime = block.timestamp + 365 days;
+    }
+
+    /**
+     * @notice Decommissions the contract. Can be called only
+     *          after a year since `start()` was invoked. Burns
+     *          all OGV held in the contract and transfers OGN
+     *          to address(1).
+     */
+    function decommission() external {
+        // Only after a year of staking
+        if (endTime == 0 || isMigrationActive()) {
+            revert MigrationNotComplete();
+        }
+
+        emit Decommissioned();
+
+        uint256 ogvBalance = ogv.balanceOf(address(this));
+        if (ogvBalance > 0) {
+            // Burn all OGV
+            ogv.burn(ogvBalance);
+        }
+
+        uint256 ognBalance = ogn.balanceOf(address(this));
+        if (ognBalance > 0) {
+            // OGN doesn't allow burning of tokens. Has `onlyOwner`
+            // modifier on `burn` and `burnFrom` methods. Also,
+            // `transfer` has a address(0) check. So, this transfers
+            // everything to address(1). The `owner` multisig of
+            // OGN token can call `burnFrom(address(1))` later.abi
+
+            ogn.transfer(address(1), ognBalance);
```

Comment:
Let's do `0x000000000000000000000000000000000000dead` instead of `0x1` Makes things clearer that this is a burn.

Reply:
Changed it to `address(0xdead)`